crates/rattler_conda_types/src/conda_lock/mod.rs

```diff
@@ -0,0 +1,222 @@
+/// Contains an enumeration for the different types of hashes for a package
+pub enum PackageHashes {
+    /// Contains an MD5 hash
+    Md5(String),
+    /// Contains as Sha256 Hash
+    Sha256(String),
+    /// Contains both hashes
+    Md5Sha256(String, String),
+}
+
+// This implementation of the `Deserialize` trait for the `PackageHashes` struct
+//
+// It expects the input to have either a `md5` field, a `sha256` field, or both.
+// If both fields are present, it constructs a `Md5Sha256` instance with their values.
+// If only the `md5` field is present, it constructs a `Md5` instance with its value.
+// If only the `sha256` field is present, it constructs a `Sha256` instance with its value.
+// If neither field is present it returns an error
+impl<'de> Deserialize<'de> for PackageHashes {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct Temp {
+            md5: Option<String>,
+            sha256: Option<String>,
+        }
+
+        let temp = Temp::deserialize(deserializer)?;
+        Ok(match (temp.md5, temp.sha256) {
+            (Some(md5), Some(sha)) => Md5Sha256(md5, sha),
+            (Some(md5), None) => Md5(md5),
+            (None, Some(sha)) => Sha256(sha),
+            _ => return Err(Error::custom("Expected `sha256` field `md5` field or both")),
+        })
+    }
+}
+
+/// Default category of a locked package
+fn default_category() -> String {
+    "main".to_string()
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct LockedDependency {
+    /// Package name of dependency
+    name: String,
+    /// Locked version
+    version: String,
+    /// Pip or Conda managed
+    manager: Manager,
+    /// What platform is this package for
+    platform: Platform,
+    /// What are its own dependencies
+    dependencies: HashMap<PackageName, VersionConstraint>,
+    /// URL to find it at
+    url: Url,
+    /// Hashes of the package
+    hash: PackageHashes,
+    /// Is the dependency optional
+    optional: bool,
+    /// Used for pip packages
+    #[serde(default = "default_category")]
+    category: String,
+    /// ???
+    source: Option<Url>,
+    /// Build string
+    build: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct DependencySource {
+    // According to:
+    // https://github.com/conda/conda-lock/blob/854fca9923faae95dc2ddd1633d26fd6b8c2a82d/conda_lock/lockfile/models.py#L27
+    // It also has a type field but this can only be url at the moment
+    // so leaving it out for now!
+    /// URL of the dependency
+    pub url: Url,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Channel {
+    /// Called `url` but can also be the name of the channel e.g. `conda-forge`
+    pub url: String,
+    pub used_env_vars: Vec<String>,
+}
+
+#[cfg(test)]
+mod test {
+    use super::PackageHashes;
+    use crate::conda_lock::CondaLock;
+    use serde_yaml::from_str;
+
+    #[test]
```

Comment:
Maybe you can add an `insta` test for a few lock files here? That would alert us in the future if something changes in the serialization/deserialization code.

Reply:
Yes, added it for one lock file. Was already working on another lock file in a different PR.